spyvm/test/test_primitives.py

```diff
@@ -492,6 +492,17 @@ def test_image_name():
     w_v = prim(primitives.IMAGE_NAME, [2])
     assert w_v.bytes == list(IMAGENAME)
 
+def test_set_image_name(monkeypatch):
+    def set_image_name(new_name):
+        assert new_name == "anImage.image" or new_name == "newImageName.image"
```

Comment:
You got me…the `prim` setup code calls `mock` which gives the image that name. So our handler is called twice (old name, then new name)